<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="phruta">
<title>Advanced methods • phruta</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Advanced methods">
<meta property="og:description" content="phruta">
<meta property="og:image" content="https://docs.ropensci.org/phruta/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">phruta</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/phruta.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/index.html">Articles</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/cromanpa" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/phruta/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Advanced methods</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/phruta/blob/HEAD/vignettes/Phruta_advanced.Rmd" class="external-link"><code>vignettes/Phruta_advanced.Rmd</code></a></small>
      <div class="d-none name"><code>Phruta_advanced.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ol style="list-style-type: decimal">
<li><a href="#paragraph0">Working with non-animal/plant groups in <code>phruta</code></a></li>
<li><a href="#paragraph1">Creating taxonomic constraints in <code>phruta</code></a></li>
<li><a href="#paragraph2">Running PartitionFinder in <code>phruta</code></a></li>
<li><a href="#paragraph3">Partitioned analyses in RAxML</a></li>
<li><a href="#paragraph4">Identifying rogue taxa</a></li>
</ol>
<div class="section level3">
<h3 id="working-with-non-animal-nor-plant-groups-in-phruta">Working with non-animal nor plant groups in <code>phruta</code> <a name="paragraph0"></a><a class="anchor" aria-label="anchor" href="#working-with-non-animal-nor-plant-groups-in-phruta"></a>
</h3>
<p>As explained in the brief intro to <code>phruta</code>, the <code><a href="../reference/sq.curate.html">sq.curate()</a></code> function is primarily designed to curate taxonomic datasets using <code>gbif</code>. Alto <code>gbif</code> is extremely fast and efficient, it is largely designed to deal with animals and plants. If you’re interested in using the <code>gbif</code> backbone for curating sequence regardless of the kingdom use the following approach:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/taxonomy.retrieve.html">taxonomy.retrieve</a></span><span class="op">(</span>species_names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis_catus"</span>, <span class="st">"PREDICTED:_Vulpes"</span>,</span>
<span>                  <span class="st">"Phoca_largha"</span>, <span class="st">"PREDICTED:_Phoca"</span> ,</span>
<span>                  <span class="st">"PREDICTED:_Manis"</span> , <span class="st">"Felis_silvestris"</span> , <span class="st">"Felis_nigripes"</span><span class="op">)</span>,</span>
<span>                  database<span class="op">=</span><span class="st">'gbif'</span>, kingdom<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p>Note that the <code>kingdom</code> argument is set to <code>NULL</code>. However, as indicated in the first vignette, <code>gbif</code> is efficient for retrieving accurate taxonomy when we provide details on the <code>kingdom</code>. Given that all the species we’re interested in are animals, we could just use:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/taxonomy.retrieve.html">taxonomy.retrieve</a></span><span class="op">(</span>species_names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis_catus"</span>, <span class="st">"PREDICTED:_Vulpes"</span>,</span>
<span>                  <span class="st">"Phoca_largha"</span>, <span class="st">"PREDICTED:_Phoca"</span> ,</span>
<span>                  <span class="st">"PREDICTED:_Manis"</span> , <span class="st">"Felis_silvestris"</span> , <span class="st">"Felis_nigripes"</span><span class="op">)</span>,</span>
<span>                  database<span class="op">=</span><span class="st">'gbif'</span>, kingdom<span class="op">=</span><span class="st">'animals'</span><span class="op">)</span></span></code></pre></div>
<p>We could also do the same for plants by using <code>plants</code> instead of <code>animals</code> in the <code>kingdom</code> argument. Now, what if we were interested in following other databases to retrieve taxonomic information for the species in our database? The latest version of <code>phruta</code> allow users to select the desired database. The databases follow the <code><a href="https://docs.ropensci.org/taxize/reference/classification.html" class="external-link">taxize::classification()</a></code> function. Options are: <code>ncbi</code>, <code>itis</code>, <code>eol</code>, <code>tropicos</code>, <code>nbn</code>, <code>worms</code>, <code>natserv</code>, <code>bold</code>, <code>wiki</code>, and <code>pow</code>. Please select only one. Note that the <code>gbif</code> option in <code><a href="https://docs.ropensci.org/taxize/reference/classification.html" class="external-link">taxize::classification()</a></code> is replaced by the internal <code>gbif</code> in <code>phruta</code>.</p>
<p>Now, let’s assume that we were interested in curating our database using <code>itis</code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/taxonomy.retrieve.html">taxonomy.retrieve</a></span><span class="op">(</span>species_names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis_catus"</span>, <span class="st">"PREDICTED:_Vulpes"</span>,</span>
<span>                  <span class="st">"Phoca_largha"</span>, <span class="st">"PREDICTED:_Phoca"</span> ,</span>
<span>                  <span class="st">"PREDICTED:_Manis"</span> , <span class="st">"Felis_silvestris"</span> , <span class="st">"Felis_nigripes"</span><span class="op">)</span>,</span>
<span>                  database<span class="op">=</span><span class="st">'itis'</span><span class="op">)</span></span></code></pre></div>
<p>Using alternative databases is sometimes desirable. Please make sure you review which the best database is for your target group is before selecting one.</p>
</div>
<div class="section level3">
<h3 id="creating-taxonomic-constraints-in-phruta">Creating taxonomic constraints in <code>phruta</code> <a name="paragraph1"></a><a class="anchor" aria-label="anchor" href="#creating-taxonomic-constraints-in-phruta"></a>
</h3>
<p>For different reasons, phylogenetic analyses sometimes require of tree constraints. <code>phruta</code> can automatically generate trees in accordance to taxonomy and a backbone topology. We divide constraint trees into two: (1) ingroup+outgroup and (2) particular clades.</p>
<div class="section level4">
<h4 id="ingroup-outgroup">ingroup + outgroup<a class="anchor" aria-label="anchor" href="#ingroup-outgroup"></a>
</h4>
<p>In this constraint type, <code>phruta</code> will create monophyletic groups for each of the taxonomic groups in the database (for selected target columns). Finally, it will generate tree with the same topology provided in the <code>Topology</code> argument. The user will provide the species names of the outgroup taxa as a vector of string that should fully match the names in the taxonomy file.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tree.constraint.html">tree.constraint</a></span><span class="op">(</span></span>
<span>                taxonomy_folder <span class="op">=</span> <span class="st">"1.CuratedSequences"</span>,</span>
<span>                targetColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kingdom"</span>, <span class="st">"phylum"</span>, <span class="st">"class"</span>, <span class="st">"order"</span>, </span>
<span>                                  <span class="st">"family"</span>, <span class="st">"genus"</span>, <span class="st">"species_names"</span><span class="op">)</span>,</span>
<span>                Topology <span class="op">=</span> <span class="st">"((ingroup), outgroup);"</span>,</span>
<span>                outgroup <span class="op">=</span> <span class="st">"Manis_pentadactyla"</span></span>
<span> <span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="IngroupOutgroup.jpg" alt=""><p class="caption">Tree constraints ingroup+outgroup (example shown based on a much limited number of genes; see the examples in the package).</p>
</div>
</div>
<div class="section level4">
<h4 id="particular-clades">Particular clades<a class="anchor" aria-label="anchor" href="#particular-clades"></a>
</h4>
<p>In this constraint type, <code>phruta</code> will create a constraint tree for particular clades. For instance, let’s assume that we only need to create a tree constraining the monophyly within two genera and their sister relationships:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tree.constraint.html">tree.constraint</a></span><span class="op">(</span> taxonomy_folder <span class="op">=</span> <span class="st">"1.CuratedSequences"</span>,</span>
<span>                 targetColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kingdom"</span>, <span class="st">"phylum"</span>, <span class="st">"class"</span>, </span>
<span>                                   <span class="st">"order"</span>, <span class="st">"family"</span>, <span class="st">"genus"</span>, <span class="st">"species_names"</span><span class="op">)</span>,</span>
<span>                 Topology <span class="op">=</span> <span class="st">"((Felis), (Phoca));"</span></span>
<span> <span class="op">)</span></span></code></pre></div>
<p>Note that the key aspect in here is the <code>Topology</code> argument. It is a newick tree.</p>
<div class="figure">
<img src="byclade.jpg" alt=""><p class="caption">Tree constraints by clade (example shown based on a much limited number of genes; see the examples in the package).</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="running-partitionfinder-in-phruta">Running PartitionFinder in <code>phruta</code> <a name="paragraph2"></a><a class="anchor" aria-label="anchor" href="#running-partitionfinder-in-phruta"></a>
</h3>
<p>With the current version of <code>phruta</code>, users are able to run <code>PartitionFinder</code> v1 within <code>R</code>. For this, users should provide the name of the folder where the alignments are stored, a particular pattern in the file names (<code>masked</code> in our case), and which models will be run in <code>PartitionFinder</code>. This function will download <code>PartitionFinder</code>, generate the input files, and run it all within R. The output files will be in a new folder within the working directory.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sq.partitionfinderv1.html">sq.partitionfinderv1</a></span><span class="op">(</span>folderAlignments <span class="op">=</span> <span class="st">"2.Alignments"</span>,</span>
<span>                    FilePatterns <span class="op">=</span> <span class="st">"Masked"</span>,</span>
<span>                    models <span class="op">=</span> <span class="st">"all"</span></span>
<span> <span class="op">)</span></span></code></pre></div>
<p>Unfortunately, the output files are not integrated with the current <code>phruta</code> pipeline. This will be part of a new release. However, users can still perform gene-based partitioned analyses within <code>RAxML</code> or can use PartitionFinder’s output files to inform their own analyses outside <code>phruta</code>.</p>
</div>
<div class="section level3">
<h3 id="partitioned-analyses-in-raxml">Partitioned analyses in RAxML <a name="paragraph3"></a><a class="anchor" aria-label="anchor" href="#partitioned-analyses-in-raxml"></a>
</h3>
<p>Users can now run partitioned analyses in <code>RAxML</code> within <code>phruta</code>. This approach is implemented by setting the <code>partitioned</code> argument in <code>tree.raxml</code> to <code>TRUE</code>. For now, partitions are based on the genes are being analyzed. The same model is used to analyze each partition. More details on partitioned analyses can be customized by passing arguments in <code><a href="https://rdrr.io/pkg/ips/man/raxml.html" class="external-link">ips::raxml</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tree.raxml.html">tree.raxml</a></span><span class="op">(</span>folder <span class="op">=</span> <span class="st">"2.Alignments"</span>, FilePatterns <span class="op">=</span> <span class="st">"Masked"</span>,</span>
<span>           raxml_exec <span class="op">=</span> <span class="st">"raxmlHPC"</span>, Bootstrap <span class="op">=</span> <span class="fl">100</span>,</span>
<span>           outgroup <span class="op">=</span> <span class="st">"Manis_pentadactyla"</span>,</span>
<span>           partitioned<span class="op">=</span><span class="cn">T</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="identifying-rogue-taxa">Identifying rogue taxa <a name="paragraph4"></a><a class="anchor" aria-label="anchor" href="#identifying-rogue-taxa"></a>
</h3>
<p><code>phruta</code> can help users run <code>RogueNaRok</code> implemented in the <code>Rogue</code> R package. Users can then examine whether rogue taxa should be excluded from the analyses. <code><a href="../reference/tree.roguetaxa.html">tree.roguetaxa()</a></code> uses the bootstrap trees generated using the <code><a href="../reference/tree.raxml.html">tree.raxml()</a></code> function along with the associated best tree to identify rogue taxa.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tree.roguetaxa.html">tree.roguetaxa</a></span><span class="op">(</span>folder <span class="op">=</span> <span class="st">"3.Phylogeny"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
